solr/core/src/java/org/apache/solr/search/stats/ExactStatsCache.java

```diff
@@ -94,6 +94,12 @@ protected ShardRequest doRetrieveStatsRequest(ResponseBuilder rb) {
   protected void doMergeToGlobalStats(SolrQueryRequest req, List<ShardResponse> responses) {
     Set<Term> allTerms = new HashSet<>();
     for (ShardResponse r : responses) {
+      if ("true".equalsIgnoreCase(req.getParams().get(ShardParams.SHARDS_TOLERANT)) && r.getException() != null) {
```

Comment:
Should we use `getBool` here instead?

Reply:
No, `SHARDS_TOLERANT` can have third parameter https://lucene.apache.org/solr/guide/8_6/solrcloud-query-routing-and-read-tolerance.html#shards-tolerant-parameter
```
In addition to true and false, shards.tolerant may also be set to requireZkConnected - see below.
```

Reply:
@Hronom I'd like to merge this PR soon, can you add a couple lines to `TestDefaultStatsCache` to verify this fix works? `BaseDistributedSearchTestCase` already creates 3 "dead servers", which you can explicitly add to the query params; `shards` parameter using its `setDistributedParams(params)` method.

Reply:
@sigram thank you for details there to put this, let me work on this, I will ping you back when it's done

Reply:
@sigram @madrob I added test that reproduces the problem in `TestExactStatsCache`.

Please can you adjust it(if needed) to nicely feet in solr tests suits.

The trick here with this issue, that is reproducible only when at least one shard is fully down. This is why I didn't use `setDistributedParams`, since it's add's one work replica, so all shards is healthy and there no situation when one shard is completely down.